File: files/2025-08-03/CSEC 03082025.py
import pandas as pd

path = "files/2025-08-03/Challenge 48.xlsx"
input = pd.read_excel(path, usecols="A:H", skiprows=1, nrows=4)
test  = pd.read_excel(path, usecols="J:K", nrows=3)

result = input.iloc[:, [0]].copy()
result['combined'] = input.iloc[:, 1:8].astype(str).agg(''.join, axis=1)\
    .str.replace(r'^0+|0+$', '', regex=True)\
    .apply(lambda x: ','.join(x) if pd.notnull(x) else x)
result.columns = ['item', 'combined']
test.columns = ['item', 'combined']

print(result.equals(test)) # True

File: files/2025-08-03/CSEC 03082025.R
library(tidyverse)
library(readxl)

path = "files/2025-08-03/Challenge 48.xlsx"
input = read_excel(path, range = "A2:H5")
test  = read_excel(path, range = "J1:K4")

result = input %>%
  unite("combined", -c(1), sep = "") %>%
  mutate(combined = str_replace_all(combined, "^0+|0+$", "")) %>%
  mutate(combined = str_replace_all(combined, "(?<=.)(?=.)", ","))

names(result) = c("item", "combined")
names(test) = c("item", "combined")

all.equal(result, test) 
# > [1] TRUE
